test/setup-after-env-global.js

```diff
@@ -0,0 +1,23 @@
+const ignoredMessages = [
```

Comment:
This approach tries to ignore the deprecation messages globally. So far, it works well. For now, I'm requiring it for the react native unit tests since I detected a lot of failures there but I will use it for others if it works well.

Reply:
I agree, but a *lot* of tests fail due to the missing expectations; I think around 100 tests if we count the ~70 in the mobile suite. Adding the expectations to each test would require a significant amount of work. Maybe we should bite the bullet and cherry-pick the other migration PRs in this branch to fix the tests and perhaps revert the hotfix?

Reply:
I don't understand why that is the case in release/17.5 but not on trunk?

Reply:
Also, I haven't checked `trunk`, it's possible that there are failing tests on `trunk` related to the recent fix. I assumed those migration PRs would also fix the consumers in GB, no? That would mean they would be fetched from the right scope and the proxy wouldn't wouldn't output the deprecation warning.

Reply:
So you're saying that 17.5 didn't include the migration for a given key "a" but we're still calling the "deprecated" call for this same key and shiming it? 

That sounds bad on multiple levels:

 - It seems that this will cause the preference to not work at all because the editor is going to try to set `core/edit-post` and read `core` (because of the proxy and trigger a deprecation).

Do you know which migrations didn't make it into 17.5? We should just remove them from the "proxy"

Reply:
Yes, 17.5 didn't include all migrations; see the description for https://github.com/WordPress/gutenberg/pull/58031, quoting:


> One solution would be to cherry-pick the missing changesets. Still, I'm not sure if there could be other consequences (i.e if those changesets depend on other changesets that are supposed to be released with 17.6), so I've come up with this hotfix that should be merged only to the release/17.5 branch.

As for:

> It seems that this will cause the preference to not work at all because the editor is going to try to set core/edit-post and read core (because of the proxy and trigger a deprecation).

That's what the 17.5-specific hotfix here aimed to fix: https://github.com/WordPress/gutenberg/pull/58031. If it can't find on `core`, it will just try to get it from whatever scope was passed originally.

Reply:
> Do you know which migrations didn't make it into 17.5? We should just remove them from the "proxy"

I don't, I can try to check today. Do we have a project or epic that encompasses all those migration PRs?

Reply:
> Do we have a project or epic that encompasses all those migration PRs?
Nevermind, I can just get the list from the array in the proxy and build a list for `release/17.5` using it. I'll get the list and remove the ones that have not been migrated in 17.5 from the proxy's list and will spin up a new 17.5-specific PR. Thanks for the insight!

Reply:
Related work continued in: https://github.com/WordPress/gutenberg/pull/58145.